build/makelib/golang.mk

```diff
@@ -221,4 +215,10 @@ $(CONTROLLER_GEN) $(YQ):
 		go get github.com/mikefarah/yq/v3;\
 		mv $$CONTROLLER_GEN_TMP_DIR/yq $(YQ) ;\
 		rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
\ No newline at end of file
+	}
+
+export CODE_GENERATOR_VERSION=0.20.0
+export CODE_GENERATOR=$(TOOLS_HOST_DIR)/code-generator-$(CODE_GENERATOR_VERSION)
+$(CODE_GENERATOR):
+	mkdir -p $(TOOLS_HOST_DIR)
+	curl -sL https://github.com/kubernetes/code-generator/archive/refs/tags/v${CODE_GENERATOR_VERSION}.tar.gz | tar -xz -C $(TOOLS_HOST_DIR)
```

Comment:
Is this because we removed `build/codegen/tools.go`?

Reply:
Yes.

Reply:
Hum I guess I'm wondering if having `build/codegen/tools.go` and the go.mod wasn't a bit easier. If we go with that approach I suspect we could go something similar to `./build/crds/crds.go` but then isn't this defeating the purpose of the go modules?

Reply:
With vendoring, `vendor/k8s.io/code-generator` is installed under the top directory. After that, we use `code-generator/generate-groups.sh`. I don't think go module mechanism is not proper way for this purpose because code-generator is not use by any go code.

One possibility is to use `${GOPATH}/pkg/dep/.../code-generator`. However, I consider it's an abuse of go mod.

Reply:
Ok, I'm fine with that.